stl/src/locale.cpp

```diff
@@ -49,29 +44,31 @@ _MRTIMP2_PURE locale __CLRCALL_PURE_OR_CDECL locale::global(const locale& loc) {
 
 #if STDCPP_IMPLIB || !defined(_M_CEE_PURE)
 // facets associated with C categories
-#define ADDFAC(Facet, cat, ptrimp, ptrloc)                                            \
-    if ((_CATMASK(Facet::_Getcat()) & cat) == 0) {                                    \
-        ;                                                                             \
-    } else if (ptrloc == nullptr) {                                                   \
-        ptrimp->_Addfac(new Facet(lobj), Facet::id);                                  \
-    } else {                                                                          \
-        ptrimp->_Addfac((locale::facet*) &_STD use_facet<Facet>(*ptrloc), Facet::id); \
+#define ADDFAC(Facet, cat, ptrimp, ptrloc)                                                                  \
+    if ((_CATMASK(Facet::_Getcat()) & cat) == 0) {                                                          \
+        ;                                                                                                   \
+    } else if (ptrloc == nullptr) {                                                                         \
+        ptrimp->_Addfac(new Facet(lobj), Facet::id);                                                        \
+    } else {                                                                                                \
+        ptrimp->_Addfac(                                                                                    \
+            const_cast<locale::facet*>(static_cast<const locale::facet*>(&_STD use_facet<Facet>(*ptrloc))), \
+            Facet::id);                                                                                     \
     }
 
-using _T1 = ctype<char>;
-using _T2 = num_get<char, _Initer>;
-using _T3 = num_put<char, _Outiter>;
-using _T4 = numpunct<char>;
-using _T5 = codecvt<char, char, _Mbstatet>;
+using _Tc1 = ctype<char>;

```

Comment:
Yuck. (no change requested)

Reply:
Yeah. We need the typedefs because of the macro usage; if we can convert this to a function template in the future, the typedefs can go away.